druid/src/widget/env_scope.rs

```diff
@@ -21,8 +21,13 @@ use tracing::instrument;
 
 /// A widget that accepts a closure to update the environment for its child.
 pub struct EnvScope<T, W> {
-    pub(crate) f: Box<dyn Fn(&mut Env, &T)>,
     pub(crate) child: WidgetPod<T, W>,
+    pub(crate) current_child_env: Option<Env>,
+    pub(crate) prev_super_env: Option<Env>,
+    pub(crate) overrides: Env,
+    pub(crate) modify_env: Option<Box<dyn Fn(&T, &mut Env)>>,
+    #[allow(clippy::type_complexity)]
+    pub(crate) should_modify_env_now: Option<Box<dyn Fn(&T, &T, &Env) -> bool>>,
```

Comment:
I would listen to clippy here and add a `type EnvInvalidationCheck = dyn Fn(&T, &T, &Env) -> bool` declaration at the top of the file.

Reply:
Ok, I wasn't even sure what to name it.

Reply:
heh me neither, just brainstorming. :)